adapter/internal/api/apis_impl.go

```diff
@@ -218,37 +271,102 @@ func updateAPI(name, version string, apiProject ProjectAPI, environments []strin
-	apiProject.APILifeCycleStatus = strings.ToUpper(data[lifeCycleStatus].(string))
-
-	var securitySchemesTypes []string = nil
-	if data[securityScheme] != nil {
-		securitySchemes := data[securityScheme].([]interface{})
-		for _, scheme := range securitySchemes {
-			securitySchemesTypes = append(securitySchemesTypes, scheme.(string))
-		}
+
+	apiProject.SecurityScheme = apiObject.Data.SecurityScheme
+
+	var apiHashValue string = generateHashValue(apiObject.Data.APIName, apiObject.Data.APIName)
+
+	endpointConfig := apiObject.Data.EndpointConfig
+
+	// production Endpoints set
+	var productionEndpoint string
+	productionEndpoint = resolveEnvValueForEndpointConfig("api_" + apiHashValue + "_prod_endpoint_0")
+	if productionEndpoint == "" {
+		productionEndpoint = endpointConfig.ProductionEndpoints.Endpoint
 	}
-	apiProject.SecurityScheme = securitySchemesTypes
-	loggers.LoggerAPI.Infof("apiProject.SecurityScheme %v", apiProject.SecurityScheme)
+	apiProject.ProductionEndpoint = productionEndpoint
+	loggers.LoggerAPI.Infof("apiProject.ProductionEndpoint %v", apiProject.ProductionEndpoint)
 
-	apiProject.endpointImplementationType = data[endpointImplementationType].(string)
+	// sandbox Endpoints set
+	var sandboxEndpoint string
+	sandboxEndpoint = resolveEnvValueForEndpointConfig("api_" + apiHashValue + "_sand_endpoint_0")
+	if sandboxEndpoint == "" {
+		sandboxEndpoint = endpointConfig.SandBoxEndpoints.Endpoint
+	}
+	apiProject.SandboxEndpoint = sandboxEndpoint
+	loggers.LoggerAPI.Infof("apiProject.SandboxEndpoint %v", apiProject.SandboxEndpoint)
+
+	// production Endpoint security
+	prodEpSecurity, _ := retrieveEndPointSecurityInfo("api_"+apiHashValue,
+		endpointConfig.EndpointSecurity.Production, "prod")
+	loggers.LoggerAPI.Infof("apiProject_security %v", endpointConfig.EndpointSecurity.Production)
+
+	// sandbox Endpoint security
+	sandBoxEpSecurity, _ := retrieveEndPointSecurityInfo("api_"+apiHashValue,
+		endpointConfig.EndpointSecurity.Sandbox, "sand")
 
-	endpointConfig := data["endpointConfig"].(map[string]interface{})
-	if endpointConfig["sandbox_endpoints"] != nil {
-		sandboxEndpoints := endpointConfig["sandbox_endpoints"].(map[string]interface{})
-		apiProject.SandboxEndpoint = sandboxEndpoints["url"].(string)
+	epSecurity := config.EndpointSecurity{
+		SandBox:    sandBoxEpSecurity,
+		Production: prodEpSecurity,
 	}
-	if endpointConfig["production_endpoints"] != nil {
-		productionEndpoints := endpointConfig["production_endpoints"].(map[string]interface{})
-		apiProject.ProductionEndpoint = productionEndpoints["url"].(string)
+	apiProject.EndpointSecurity = epSecurity
+}
+
+func generateHashValue(apiName string, apiVersion string) string {
+	endpointConfigSHValue := sha1.New()
+	endpointConfigSHValue.Write([]byte(apiName + ":" + apiVersion))
+	return hex.EncodeToString(endpointConfigSHValue.Sum(nil)[:])
+}
+
+func resolveEnvValueForEndpointConfig(envKey string) string {
```

Comment:
Would be more convenient if you could pass the default value here as well? 
Coz you can avoid the later check for empty string

Reply:
Fixed with https://github.com/wso2/product-microgateway/pull/1785/commits/5e7db9dba2577979211d7713f6690403ca650326